src/ReadComicsTo/Parser.ts

```diff
@@ -0,0 +1,170 @@
+import {Chapter, LanguageCode, Manga, MangaStatus, MangaTile, Tag, TagSection} from 'paperback-extensions-common'
+
+const READCOMICTO_DOMAIN = 'https://readcomiconline.to'
+
+export class Parser {
+
+    parseMangaDetails($: CheerioSelector, mangaId: string): Manga {
+    
+
+    let titles =  [$('.bigChar', $('.bigBarContainer').first()).text().trim()]
+    let url = $('img', $('.rightBox')).attr('src')
+    let image = url?.includes('http') ? url : `${READCOMICTO_DOMAIN}${url}`
+    
+
+
+    let status = MangaStatus.ONGOING, author, released, rating: number = 0,artist, views,summary
+
+    let tagArray0 : Tag[] = []
+    let i = 0
+    let infoElement = $("div.barContent").first()
+    artist =this.decodeHTMLEntity($('p:has(span:contains(Artist:)) > a', infoElement).first().text())
+    author = ($('p:has(span:contains(Writer:)) > a', infoElement).first().text())
+    summary = ($('p:has(span:contains(Summary:)) ~ p', infoElement).text())
+    released = this.decodeHTMLEntity($('p:has(span:contains(Publication date:))', infoElement).first().text()).replace('Publication date:', '').trim()
+    
+    let statusViewsParagraph = $('p:has(span:contains(Status:))', infoElement).first().text().toLowerCase()
+    status = statusViewsParagraph.includes('ongoing') ? MangaStatus.ONGOING : MangaStatus.COMPLETED
+    views = Number(statusViewsParagraph.replace('\n', '').split('\n')[1]?.trim()?.replace(/\D/g, ''))
+
+    let genres = $('p:has(span:contains(Genres:)) > a', infoElement).toArray()
+    for (let obj of genres) {
+        let id = $(obj).attr('href')
+        let label = this.decodeHTMLEntity($(obj).text().trim())
+        if (typeof id === 'undefined' || typeof label === 'undefined') continue
+        tagArray0 = [...tagArray0, createTag({id: id, label: label})]
+    }
+    let tagSections: TagSection[] = [createTagSection({ id: '0', label: 'genres', tags: tagArray0 })]
+      return createManga({
+        id: mangaId,
+        rating: rating,
+        titles: titles,
+        image: image ?? '',
+        status: status,
+        author: this.decodeHTMLEntity(author ?? ''),
+        artist: artist,
+        views: views,
+        tags: tagSections,
+        desc: this.decodeHTMLEntity(summary ?? ''),
+        lastUpdate: released
+      })
+    }
+   
+
+    parseChapterList($: CheerioSelector, mangaId: string) : Chapter[] { 
+    
+    let chapters: Chapter[] = []
+
+    let chapArray = $('tr', $('.listing').first()).toArray().reverse()
+    for(let i = 0; i < chapArray.length; i++) {
+           let obj = chapArray[i]
+           let chapterId = $('a', obj)?.first().attr('href')?.replace(`/Comic/${mangaId}/`, '')
+           let chapter = $('td', obj).first()
+           let chapNum = i + 1
+           let chapName = chapter.text().trim()
+           let time = $('td', $(obj)).last().text().trim()
+           if (typeof chapterId === 'undefined') continue
```

Comment:
I suggest expanding this a little more. There are some elements in your `createChapter({..})` method which is contingent on the data you get up here.

```suggestion
           if (typeof chapterId === 'undefined' || !time || isNaN(chapNum)) continue
```

Or something along these lines. Any chance you can avoid potentially giving the app bad data is a good thing. The app MIGHT handle some bad data, but.. Well. It also might not.

Reply:
Note that I haven't checked if the `isNaN(chapNum)` is valid syntax. I believe it is, but I didn't have an IDE to back that up.